mixer/cmd/build.go

```diff
@@ -199,113 +213,163 @@ var buildFormatNewCmd = &cobra.Command{
-		if err = helpers.RunCommandSilent("cp", "-al", source, dest); err != nil {
-			failf("Failed to copy +20 chroots to +10: %s\n", err)
+		// Write the +0 version to LAST_VER so that we reference in both +10 and +20 manifests as the 'previous:'
+		if err = ioutil.WriteFile(filepath.Join(b.Config.Builder.ServerStateDir, "image", "LAST_VER"), []byte(strconv.Itoa(ver-20)), 0644); err != nil {
+			failf("Couldn't update LAST_VER file: %s", err)
 		}
+	},
+}
 
-		// Copy the old groups.ini file back which contains ALL original bundle names
-		// to account for any removed bundles in this build when creating manifests
-		fmt.Println(" Copying full groups.ini back to working directory")
-		if err = b.RevertFullGroupsINI(); err != nil {
+// This is the first build in the new format. The content is the same as the +10
+// but the manifests might be created differently if a new manifest template is
+// defined for the new format.
+var buildFormatNewCmd = &cobra.Command{
+	Use:   "new",
+	Short: "Build the +20 version in the new format for the format bump",
+	Long:  `Build the +20 version in the new format for the format bump`,
+	Run: func(cmd *cobra.Command, args []string) {
+		b, err := builder.NewFromConfig(configFile)
+		if err != nil {
 			fail(err)
 		}
 
-		// Set the format back to the previous format version before building the +10 update
-		prevFormat, err := strconv.Atoi(b.State.Mix.Format)
+		// Set mixversion to +20 build now, we only need to build update since we
+		// already have the bundles created
+		ver, err := strconv.Atoi(b.MixVer)
 		if err != nil {
 			fail(err)
 		}
-		prevFormat--
-		if err = b.UpdateFormatVersion(strconv.Itoa(prevFormat)); err != nil {
-			fail(err)
+		if err = b.UpdateMixVer(ver + 10); err != nil {
+			failf("Couldn't update Mix Version")
 		}
-		// Set mixversion to the +10 since we have used +20 up to this point
-		if err = b.UpdateMixVer(prevVersion); err != nil {
+
+		// Set format to format+1 for future builds
+		if err = b.UpdateFormatVersion(buildFlags.newFormat); err != nil {
 			fail(err)
 		}
-		// Since we build +10 out of order, restore LAST_VER to the previous version
-		if err := ioutil.WriteFile(filepath.Join(b.Config.Builder.ServerStateDir, "image/LAST_VER"), []byte(strconv.Itoa(ver-20)), 0644); err != nil {
-			fail(err)
+
+		setWorkers(b)
+
+		// Remove the bundles from groups.ini and mixbundles so they are not
+		// tracked anymore and manifests do not get created for them
+		f := func(bundle string) error {
+			var groups []byte
+			filename := filepath.Join(b.Config.Builder.ServerStateDir, "groups.ini")
+			groups, err = ioutil.ReadFile(filename)
+			if err != nil {
+				return err
+			}
+			groupsini := string(groups)
+
+			bundleToRemove := fmt.Sprintf("\\[%s\\]\ngroup=%s", bundle, bundle)
+			re := regexp.MustCompile(bundleToRemove)
+			groupsini = re.ReplaceAllString(groupsini, "")
+
+			// We are rarely removing more than a couple bundles, so it is ok
```

Comment:
ha.

my code is slow but is ok i promis

Reply:
shush

Reply:
you know, you could just pass this a slice of strings and modify the caller (ModifyBundles) to utilize that
```go
func(bundles string...) error {
}
```
Move the matching logic into the `action` func itself so it can handle multiple matches. Then you don't have to write the file a ton of times.